### ISSUE TITLE
Hyperlinks for content types in docs are broken.

### ISSUE THREAD

**Issue by freakboy3742** [beeware/toga#2639]
### Describe the bug

Constructors for types like `Icon` and `Image` are annotated to take `IconContentT` and `ImageContentT` types. However, the places where these types are used (e.g. in the constructors for `ImageView`), the hyperlinks are broken.

### Steps to reproduce

View the documentation for [`ImageView`](https://toga.readthedocs.io/en/stable/reference/api/widgets/imageview.html#toga.ImageView)

### Expected behavior

`ImageContentT` should be a hyperlink, rather than a static link.

### Screenshots

<img width="837" alt="Screenshot 2024-06-11 at 2 56 02 PM" src="https://github.com/beeware/toga/assets/37345/76fa302b-680c-4c82-aacf-7a222cfbcc5e">

### Environment

- Operating System: All
- Python version: All
- Software versions:
  - Toga: 0.4.5

### Logs

```

```

### Additional context

The current workaround is to explicitly reference the type with `:any:ImageContentT` in the docstring; this works but is not ideal.

I think the same problem exists for *any* type alias used as an argument (StyleT, NumberInputT, IconContentT,...). This may highlight the underlying issue - the need to be able to document type aliases in Sphinx.

I've only provided a screenshot for `ImageContentT`; if the fix is generic, presumably it will correct any type that has documentation.

**Comment by rmartin16** [comment]
> I think the same problem exists for _any_ type alias used as an argument (StyleT, NumberInputT, IconContentT,...). This may highlight the underlying issue - the need to be able to document type aliases in Sphinx.

For future readers, one way to address the display of `TypeAlias` annotations in Sphinx is [sphinx-autodoc-typehints](https://github.com/tox-dev/sphinx-autodoc-typehints); this extension replaces the `TypeAlias` in the argument annotation with the actual type assigned to the alias, for instance:

![image](https://github.com/beeware/toga/assets/387848/35306854-1013-490d-949c-f156b7ed7f63)

However, this extension was [removed](https://github.com/beeware/toga/pull/2259#issuecomment-1851409730) from Toga in c13130e due to other compromises.

**Comment by freakboy3742** [comment]
> However, this extension was [removed](https://github.com/beeware/toga/pull/2259#issuecomment-1851409730) from Toga in [c13130e](https://github.com/beeware/toga/commit/c13130e45ea16894c929eea92fd2e8597f952cfe) due to other compromises.

As I recall, it was removed because `Optional[TypeVar(StyleT, bound=BaseStyle)])`, with a link to the upstream docs for `TypeVar` is moon language for the purposes we need it for. We need it to say `Optional[StyleT]`, and link to *our* documentation for the `StyleT` type.

**Comment by rmartin16** [comment]
Yeah...`sphinx-autodoc-typehints` is a nice _quick_ solution to documenting a `TypeAlias`, `TypeVar`, etc...but it quickly starts to breakdown with more complex types.

**Comment by kattni** [comment]
I lost a couple of hours digging into this. Sphinx itself now [supports documenting type aliases](https://github.com/sphinx-doc/sphinx/pull/11989). However, it seems your documentation uses `autodoc` which still [does not support them](https://github.com/sphinx-doc/sphinx/issues/10785). As suggested [here](https://github.com/sphinx-doc/sphinx/issues/10785#issuecomment-2689605528), there are no immediate plans to implement this. `autodoc` implementation was mentioned in the above-linked Sphinx PR, but was deferred. 

There are a couple of gnarly looking workarounds listed in the `autodoc` link above, written by individuals. I don't know how desperate you are to have this working; based on the lack of activity on this, I imagine not janky-workaround-desperate. But you're welcome to take a look at them and see if anything interests you.

**Comment by mhsmith** [comment]
This was fixed when switching to MkDocs.
